apps/cowswap-frontend/src/modules/trade/hooks/useNotifyWidgetTrade.ts

```diff
@@ -0,0 +1,65 @@
+import { useEffect } from 'react'
+
+import { NATIVE_CURRENCY_ADDRESS } from '@cowprotocol/common-const'
+import { getIsNativeToken } from '@cowprotocol/common-utils'
+import { CowEvents, OnTradeParamsPayload } from '@cowprotocol/events'
+import { TokenInfo, UiOrderType } from '@cowprotocol/types'
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
+
+import { EVENT_EMITTER } from 'eventEmitter'
+
+import { useDerivedTradeState } from './useDerivedTradeState'
+import { TradeType } from './useTradeTypeInfo'
+
+import { TradeDerivedState } from '../types/TradeDerivedState'
+
+export function useNotifyWidgetTrade() {
+  const { state } = useDerivedTradeState()
+
+  useEffect(() => {
+    if (!state?.tradeType) return
+
+    EVENT_EMITTER.emit(CowEvents.ON_CHANGE_TRADE_PARAMS, getTradeParamsEventPayload(state.tradeType, state))
```

Comment:
Curious, how often does this trigger?
Because of the dependency on the `state` object, might be quite often, no?

Reply:
<img width="1421" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/5c9bbfcd-1fbd-44d4-accd-9d86c32a9f61">

At the page loading it fires 3 times (see outputCurrencyAmount):
 - no quote
 - fast quote
 - optimal quote

Reply:
So, I can't say it triggers too frequent